docs/pages/guides/using-sentry.mdx

```diff
@@ -16,20 +18,212 @@ import { CODE } from '~/ui/components/Text';
+});
+```
+
+Now wrap the root component of your app with Sentry. This may be **App.js** or **index.js** depending on your project, or in [**app/\_layout.js** with Expo Router](https://docs.expo.dev/router/layouts/#create-a-layout-route).
+
+```js
+import * as Sentry from '@sentry/react-native';
+
+// Your App component here
+
+export default Sentry.wrap(App);
+```
+
+</Step>
+
+<Step label="6">
+
+### Verify the configuration
+
+Create a new release build of your app and verify that it uploads source maps correctly. You may want to add a button in your app to test that it is working and sourcemaps are wired up as expected, for example:
+
+```jsx
+import { Button } from 'react-native';
+
+// Inside some component
+<Button title="Press me" onPress={() => throw new Error('Hello, Sentry!')} />;
+```
+
+</Step>
+
+## Usage with EAS Build
+
+Ensure that `SENTRY_AUTH_TOKEN` is set in your build environment, and Sentry will automatically upload source maps for you. If you use environment variables rather than properties in your app config, ensure that those are set as well.
+
+Using the above instructions, no additional work is needed to integrate Sentry into your project when using EAS Build.
+
+## Usage with EAS Update
+
+After running `eas update`, upload the source maps to Sentry:
+
+<Terminal
+  cmd={[
+    '# Pass in the `dist` directory generated by `eas update` to the upload script',
+    '$ npx sentry-expo-upload-sourcemaps dist',
+  ]}
+/>
+
+That's it! Errors for your updates will now be properly symbolicated in Sentry.
+
+<Collapsible summary="Do you want to publish an update and the sourcemaps in one command?">
+
+You can chain the commands together with `&&` to publish an update and upload the sourcemaps in one step:
+
+<Terminal cmd={['$ eas update --channel <channel> && npx sentry-expo-upload-sourcemaps dist']} />
+
+</Collapsible>
+
+<Collapsible summary="Do you want to append additional update-related metadata to error reports?">
+
+Configuring Sentry to tag your scope with information about your update allows to see errors happening on certain updates in Sentry dashboard.
+
+Add the following snippet in the global scope as early as possible in your application's lifecycle.
+
+```js
+import * as Updates from 'expo-updates';
+
+const { metadata, extra } = Updates.manifest;
+
+Sentry.init({
+  // dsn, release, dist, etc...
+});
+
+Sentry.configureScope(scope => {
+  scope.setTag('expo-update-group-id', metadata?.updateGroup);
+  scope.setTag('expo-update-id', Updates.updateId);
+  // This will be `true` if the update is the one embedded in the build, and not one downloaded from the updates server.
+  scope.setTag('expo-is-embedded-update', Updates.isEmbeddedLaunch);
+  scope.setTag(
+    'expo-update-debug-url',
+    Updates.isEmbeddedLaunch
+      ? 'not applicable for embedded updates'
+      : `https://expo.dev/accounts/${extra?.expoClient?.owner}/projects/${extra?.expoClient?.slug}/updates/${metadata?.updateGroup}`
+  );
+});
```

Comment:
```suggestion
import * as Updates from 'expo-updates';

const { metadata, extra } = Updates.manifest;

Sentry.init({
  // dsn, release, dist, etc...
});

Sentry.configureScope(scope => {
  scope.setTag('expo-update-group-id', metadata?.updateGroup);
  scope.setTag('expo-update-id', Updates.updateId);
  // This will be `true` if the update is the one embedded in the build, and not one downloaded from the updates server.
  scope.setTag('expo-is-embedded-update', Updates.isEmbeddedLaunch);
  scope.setTag(
    'expo-update-debug-url',
    Updates.isEmbeddedLaunch
      ? 'not applicable for embedded updates'
      : `https://expo.dev/accounts/${extra?.expoClient?.owner}/projects/${extra?.expoClient?.slug}/updates/${metadata?.updateGroup}`
  );
});
```
```suggestion
import * as Updates from 'expo-updates';

const manifest = Updates.manifest;
const metadata = "metadata" in manifest ? manifest.metadata : undefined;
const extra = "metadata" in manifest ? manifest.extra : undefined;
const updateGroup =
  metadata && "updateGroup" in metadata ? metadata.updateGroup : undefined;
  
Sentry.init({
  // dsn, release, dist, etc...
});

Sentry.configureScope((scope) => {
  if (typeof updateGroup === "string") {
    scope.setTag("expo-update-group-id", updateGroup);
  }
  scope.setTag("expo-update-id", Updates.updateId);
  // This will be `true` if the update is the one embedded in the build, and not one downloaded from the updates server.
  scope.setTag("expo-is-embedded-update", Updates.isEmbeddedLaunch);
  if (Updates.isEmbeddedLaunch) {
    scope.setTag(
      "expo-update-debug-url",
      "not applicable for embedded updates"
    );
  } else if (typeof updateGroup === "string") {
    const owner = extra?.expoClient?.owner ?? "[account]";
    const slug = extra?.expoClient?.slug ?? "[project]";
    scope.setTag(
      "expo-update-debug-url",
      `https://expo.dev/accounts/${owner}/projects/${slug}/updates/${updateGroup}`
    );
  }
});
```